turbinia/workers/__init__.py

```diff
@@ -387,6 +398,14 @@ def run_wrapper(self, evidence):
     details from exceptions in the worker cannot be propogated back to the
     Turbinia TaskManager.
 
+    This method should handle (in no particular order):
+      - Exceptions thrown from run()
+      - Verifing valid TurbiniaTaskResult object is returned
+          - Check for bad results (non TurbiniaTaskResults) returned from run()
+          - Auto-close results that haven't been closed
+          - Verifying that the results are serializeable
+      - Locking to make sure only one task is active at a time
```

Comment:
I can't find the locking code here? Or am I missing something? :)

Reply:
Heh, yeah, that's what it *should* do, ;).   I have a separate PR I'm working on for this, but I just added a TODO for it in the meantime.